src/features/Conversations/ChatsTree.tsx

```diff
@@ -0,0 +1,443 @@
+          overId: String(currentId),
+        });
+      }
+
+      document.body.style.setProperty('cursor', 'grabbing');
+    };
+
+    const handleDragMove = ({ delta }: DragMoveEvent) => {
+      setOffsetLeft(delta.x);
+    };
+
+    const handleDragOver = ({ over }: DragOverEvent) => {
+      setOverId(over?.id?.toString() ?? null);
+    };
+
+    const resetState = () => {
+      setOverId(null);
+      setActiveId(null);
+      setOffsetLeft(0);
+      setCurrentPosition(null);
+
+      document.body.style.setProperty('cursor', '');
+    };
+
+    const handleDragEnd = ({ active, over }: DragEndEvent) => {
+      resetState();
+      if (projected && over) {
+        const { depth, parentId, parentType, currentType } = projected;
+        if (parentType === 'chat') {
+          return;
+        }
+        if (currentType === 'folder' && parentType === 'folder') {
+          return;
+        }
+
+        const clonedItems: FlattenedItem[] = JSON.parse(
+          JSON.stringify(flattenedTree),
+        );
+        const overIndex = clonedItems.findIndex(({ id }) => id === over.id);
+        const activeIndex = clonedItems.findIndex(({ id }) => id === active.id);
+        const activeTreeItem = clonedItems[activeIndex];
+        clonedItems[activeIndex] = { ...activeTreeItem, depth, parentId };
+        const sortedItems = arrayMove(clonedItems, activeIndex, overIndex);
+        const newItems = buildTree(sortedItems);
+        dispatch(updateChatTree({ chatTree: newItems }));
+      }
+    };
+
+    const handleDragCancel = () => {
+      resetState();
+    };
+
+    const handleRemove = (id: string) => {
+      dispatch(deleteChatTreeItem({ chatTreeId: id }));
+    };
+
+    const handleCollapse = (id: string) => {
+      dispatch(collapseChatTreeItem({ chatTreeId: id }));
+    };
+
+    return (
+      <DndContext
+        sensors={sensors}
+        collisionDetection={closestCenter}
+        measuring={measuring}
+        onDragStart={handleDragStart}
+        onDragMove={handleDragMove}
+        onDragOver={handleDragOver}
+        onDragEnd={handleDragEnd}
+        onDragCancel={handleDragCancel}
+      >
+        <SortableContext
+          items={sortedIds}
+          strategy={verticalListSortingStrategy}
+        >
+          {flattenedItems.map(
+            ({ id, name, type, children, collapsed, depth }) => (
+              <SortableTreeItem
+                key={id}
+                id={id}
+                value={id}
+                name={name}
+                type={type}
+                depth={id === activeId && projected ? projected.depth : depth}
+                indentationWidth={indentationWidth}
+                indicator={indicator}
+                collapsed={Boolean(collapsed && children.length)}
+                onCollapse={
+                  collapsible && children.length
+                    ? () => {
+                        handleCollapse(id);
+                      }
```

Comment:
create const handleCollapse = (id) => () => {}
and use like this collapsible && children.length
                    ? handleCollapse(id) : undefined